sstables/sstable_directory.cc

```diff
@@ -196,7 +220,13 @@ void sstable_directory::validate(sstables::shared_sstable sst, process_flags fla
 }
 
 future<sstables::shared_sstable> sstable_directory::load_sstable(sstables::entry_descriptor desc, sstables::sstable_open_config cfg) const {
-    auto sst = _manager.make_sstable(_schema, *_storage_opts, desc.generation, _state, desc.version, desc.format, gc_clock::now(), _error_handler_gen);
+    shared_sstable sst;
+    if (desc.prefix) {
```

Comment:
The `desc.prefix` is exactly the same as `_storage_options->s3::prefix`, so it's better to use it

Reply:
no, they are different. `desc.prefix` is a per-sstable prefix, while the latter is a per-table configuration.